firestore/query.go

```diff
@@ -364,6 +366,97 @@ func (q Query) Deserialize(bytes []byte) (Query, error) {
 	return q.fromProto(&runQueryRequest)
 }
 
+// DistanceMeasure is the distance measure to use when comparing vectors.
+type DistanceMeasure int32
+
+const (
+	// Measures the EUCLIDEAN distance between the vectors. See
+	// [Euclidean](https://en.wikipedia.org/wiki/Euclidean_distance) to learn
+	// more
+	DistanceMeasureEuclidean DistanceMeasure = 1
+
+	// Compares vectors based on the angle between them, which allows you to
+	// measure similarity that isn't based on the vectors magnitude.
+	// We recommend using DOT_PRODUCT with unit normalized vectors instead of
+	// COSINE distance, which is mathematically equivalent with better
+	// performance. See [Cosine
+	// Similarity](https://en.wikipedia.org/wiki/Cosine_similarity) to learn
+	// more.
+	DistanceMeasureCosine DistanceMeasure = 2
+
+	// Similar to cosine but is affected by the magnitude of the vectors. See
+	// [Dot Product](https://en.wikipedia.org/wiki/Dot_product) to learn more.
+	DistanceMeasureDotProduct DistanceMeasure = 3
+)
+
+// FindNearestOpts is options to use while building FindNearest vector query
+type FindNearestOpts struct {
+	Limit   int
+	Measure DistanceMeasure
+}
+
+// VectorQuery represents a vector query
+type VectorQuery struct {
+	Query
+}
+
+// FindNearest returns a query that can perform vector distance (similarity) search with given parameters.
+//
+// The returned query, when executed, performs a distance (similarity) search on the specified
+// 'vectorField' against the given 'queryVector' and returns the top documents that are closest
+// to the 'queryVector;.
+//
+// Only documents whose 'vectorField' field is a Vector of the same dimension as 'queryVector'
+// participate in the query, all other documents are ignored.
+//
+// The 'vectorField' argument can be a single field or a dot-separated sequence of
+// fields, and must not contain any of the runes "˜*/[]".
+func (q Query) FindNearest(vectorField string, queryVector VectorType, options FindNearestOpts) VectorQuery {
+	vq := VectorQuery{
```

Comment:
s/VectorType/Vector/

I think we can just return `Query` here. Why do we need `VectorQuery`?

Reply:
We don't want to allow the methods such as StartAt, EndAt, etc on VectorQuery
This design is inline with Python and Node representation.